spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -295,27 +302,69 @@ public T run(TransactionContext transaction) throws Exception {
 
 	private ResultSet executeRead(String tableName, KeySet keys, Iterable<String> columns,
 			SpannerReadOptions options) {
+		StringBuilder logSb = new StringBuilder(LOGGER.isDebugEnabled()
+				? "Executing read on table " + tableName + " with keys: " + keys
+						+ " and columns: "
+				: "Executing Read. Debug mode log not enabled.");
+		if (LOGGER.isDebugEnabled()) {
+			StringJoiner sj = new StringJoiner(",");
+			columns.forEach(col -> sj.add(col));
+			logSb.append(sj.toString());
+		}
 		if (options == null) {
+			LOGGER.debug(logSb.toString());
 			return getReadContext().read(tableName, keys, columns);
 		}
 		else {
-			ReadContext readContext = (options.hasTimestamp()
-					? getReadContext(options.getTimestamp())
-					: getReadContext());
+			ReadContext readContext;
+			if (options.hasTimestamp()) {
+				readContext = getReadContext(options.getTimestamp());
+				if (LOGGER.isDebugEnabled()) {
+					logSb.append(" at timestamp " + options.getTimestamp());
+				}
```

Comment:
this is functionally okay, but became very hard to read, can you please refactor?

Reply:
Done